kazoo/utils.py

```
@@ -0,0 +1,19 @@
+from types import MethodType
+
+
+def _on_demand(cls):
+    def rebind(dct):
+        for name, obj in dct.items():
+            try:
+                wrap, obj = obj
+                if wrap is MethodType:
+                    setattr(cls, name, wrap(obj, None, cls))
+            except (TypeError, ValueError):
```

Would it be  better to just let this raise (instead of silencing errors?)